Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -112,23 +117,36 @@ public void Initialize()
             {
                 this.Wallets.Add(wallet);
             }
-            
+
             // load data in memory for faster lookups
             this.LoadKeysLookup();
 
             // find the last chain block received by the wallet manager.
             this.WalletTipHash = this.LastReceivedBlockHash();
 
             // save the wallets file every 5 minutes to help against crashes.
-            this.asyncLoopFactory.Run("wallet persist job", token =>
-                {
-                    this.SaveToFile();
-                    this.logger.LogInformation($"Wallets saved to file at {DateTime.Now}.");
-                    return Task.CompletedTask;
-                },
-                this.nodeLifetime.ApplicationStopping,
-                repeatEvery: TimeSpan.FromMinutes(WalletSavetimeIntervalInMinutes),
-                startAfter: TimeSpan.FromMinutes(WalletSavetimeIntervalInMinutes));
+            this.asyncLoop = this.asyncLoopFactory.Run("wallet persist job", token =>
+            {
+                this.SaveToFile();
+                this.logger.LogInformation($"Wallets saved to file at {DateTime.Now}.");
+                return Task.CompletedTask;
+            },
+            this.nodeLifetime.ApplicationStopping,
+            repeatEvery: TimeSpan.FromMinutes(WalletSavetimeIntervalInMinutes),
+            startAfter: TimeSpan.FromMinutes(WalletSavetimeIntervalInMinutes));
+        }
+
+        /// <inheritdoc />
+        public void Stop()
+        {
+            if (this.asyncLoop != null)
+                this.asyncLoop.Dispose();
+
+            // safely persist the wallets to the file system before disposing
+            foreach (var wallet in this.Wallets)
```

Comment:
Can you not just call the single method `  this.SaveToFile();` same as in the loop?

Reply:
You are correct, we can just call this.SaveToFile(). I renamed them to SaveWallets and SaveWallet.. its more descriptive ;)